projectgenerator/libqgsprojectgen/dbconnector/gpkg_connector.py

```diff
@@ -26,11 +26,46 @@
 class GPKGConnector(DBConnector):
     def __init__(self, uri, schema):
         DBConnector.__init__(self, uri, schema)
-        self.conn = qgis.utils.spatialite_connect(uri)
+        self.conn = self._get_connection(uri)
         self.conn.row_factory = sqlite3.Row
         self._bMetadataTable = self._metadata_exists()
         self._tables_info = self._get_tables_info()
 
+    def _get_connection(self, uri):
+        '''
+        Get a dbapi2.Connection to GeoPackage.
+
+        Heavily based on QGIS qgis.utils.spatialite_connect, but reimplemented
+        to avoid a Windows crash using pyspatialite with GeoPackage.
+
+        See https://github.com/qgis/QGIS/blob/master/python/utils.py#L591
+        '''
+        con = sqlite3.dbapi2.connect(uri)
+        con.enable_load_extension(True)
+        cur = con.cursor()
+        libs = [
+            # SpatiaLite >= 4.2 and Sqlite >= 3.7.17, should work on all platforms
+            ("mod_spatialite", "sqlite3_modspatialite_init"),
+            # SpatiaLite >= 4.2 and Sqlite < 3.7.17 (Travis)
+            ("mod_spatialite.so", "sqlite3_modspatialite_init"),
+            # SpatiaLite < 4.2 (linux)
+            ("libspatialite.so", "sqlite3_extension_init")
+        ]
+        found = False
+        for lib, entry_point in libs:
+            try:
+                cur.execute("select load_extension('{}', '{}')".format(lib, entry_point))
+            except sqlite3.OperationalError:
+                continue
+            else:
+                found = True
+                break
+        if not found:
+            print("Cannot find any suitable spatialite module")
```

Comment:
Should raise an exception I guess.

Reply:
Yeap, [d1c614a](https://github.com/opengisch/projectgenerator/pull/112/commits/d1c614a4dcf4bafe2b1a280d613a57d783509bf3) is a temporary solution for the QGIS crash on Windows. We actually don't need the SpatiaLite module for what we are doing. Just raw SQLite queries are fine for us. However, of course fixing this in QGIS is the way to go, and then such commit can be reverted.